chain/src/chain.rs

```diff
@@ -275,10 +279,12 @@ impl Chain {
 	}
 
     /// Gets the block header by the provided output commitment
-    pub fn get_block_header_by_output_commit(&self, commit: &Commitment) -> Result<BlockHeader, Error> {
-        self.store.get_block_header_by_output_commit(commit).map_err(
-            &Error::StoreErr,
-        )
+    pub fn get_block_header_by_output_commit(&self, commit: &Commitment) -> Option<BlockHeader> {
```

Comment:
I'm not a huge fan of the error suppression here. Consumers should know when an output commitment is absent (i.e. spent) vs a data store read error. Errors can be converted by the adapter for the pool. I understand you probably try to follow the signature of get_unspent but that 's not something I had thought through for too long (meant it as temporary :-))